gen2_demo.py

```diff
@@ -123,6 +151,27 @@ def create_nn_pipeline(self, p, nodes):
             xin.setStreamName(self.input_name)
             xin.out.link(nn.input)
             setattr(nodes, self.input_name, xout)
+        elif self.source == "right": # Use spatial information
+            nodes.xout_right = p.createXLinkOut()
+            nodes.xout_right.setStreamName("right")
```

Comment:
I would separate this change from this function. This one is meant to serve only as a nn creator, so it should only have calls related to creating NN. 
Right here, both right xout and image manip are created, which should be moved to a different function.
The flow should look more like

- user provides params
- we resolve these params to what we need (so if he wants depth, we need to enable left and right camera flags, also the depth flag. In case we would like also nn, we also enable right_nn_camera flag and create nn with source=right.
- then, we do a pipeline functions based on a calls, so it would look like
  ```
  if enable_left_cam:
      pipelineManager.createLeftCam()
  if enable_right_cam:
      pipelineManager.createRightCam()
  ```
My fear is that we will get quickly tighten up by putting up too much things in a single place